client/src/Annotator/Paper/Tool/useWand.tsx

```diff
@@ -0,0 +1,157 @@
+/**
+ * Tool for selecting area around specific point ( based on colors )
+ */
+import { useState, useRef, useEffect, useCallback } from 'react';
+import paper from 'paper';
+import * as MagicWand from 'magic-wand-tool';
+
+import { Maybe, MouseEvent, ImageSize } from '../../annotator.types';
+
+import * as CONFIG from '../../annotator.config';
+
+// interfaces
+interface IToolWand {
+    (
+        isActive: boolean,
+        imageSize: Maybe<ImageSize>,
+        imageData: Maybe<ImageData>,
+        unite: (path: paper.Path) => void,
+        subtract: (path: paper.Path) => void,
+    ): ToolWandResponse;
+}
+export interface ToolWandResponse {
+    threshold: number;
+    blur: number;
+    setThreshold: (value: number) => void;
+    setBlur: (value: number) => void;
+}
+
+interface Settings {
+    threshold: number;
+    blur: number;
+}
+
+export const useWand: IToolWand = (
+    isActive,
+    imageSize,
+    imageData,
+    unite,
+    subtract,
+) => {
+    const toolRef = useRef<Maybe<paper.Tool>>(null);
+
+    const [settings, _setSettings] = useState<Settings>({
+        threshold: CONFIG.TOOLS_WAND_INITIAL_THRESHOLD,
+        blur: CONFIG.TOOLS_WAND_INITIAL_BLUR,
+    });
+
+    // private actions
+    const setThreshold = useCallback((value: number) => {
+        _setSettings(oldState => ({ ...oldState, threshold: value }));
+    }, []);
+
+    const setBlur = useCallback((value: number) => {
+        _setSettings(oldState => ({ ...oldState, blur: value }));
+    }, []);
+
+    const _flood = useCallback(
+        (
```

Comment:
if function takes so many arguments it should take object as an argument to use the named params.